src/services/LogFileManager/index.ts

```diff
@@ -202,21 +205,40 @@ class LogFileManager {
         fileData: Uint8Array,
         decoderOptions: DecoderOptions
     ): Promise<Decoder> {
-        let decoder: Decoder;
-        if (fileName.endsWith(".jsonl")) {
-            decoder = new JsonlDecoder(fileData, decoderOptions);
-        } else if (fileName.endsWith(".clp.zst")) {
-            decoder = await ClpIrDecoder.create(fileData, decoderOptions);
-        } else {
-            throw new Error(`No decoder supports ${fileName}`);
-        }
-
         if (fileData.length > MAX_V8_STRING_LENGTH) {
             throw new Error(`Cannot handle files larger than ${
                 formatSizeInBytes(MAX_V8_STRING_LENGTH)
             } due to a limitation in Chromium-based browsers.`);
         }
 
+        let decoder: Decoder;
+
+        const checkFileExtension = async () => {
+            if (fileName.endsWith(".jsonl")) {
```

Comment:
For every new extension added, we will need to specify an addition if-statement. Can we refactor the mapping into an array of config objects for example?